### ISSUE TITLE
CSI volumes links on the job page lead to a 404

### ISSUE THREAD

**Issue by apollo13** [hashicorp/nomad#9249]
### Nomad version
0.12.5

### Issue

I managed to mount a CSI volume and it shows up properly under the job volume requirements in the UI:
![image](https://user-images.githubusercontent.com/81547/97894801-3a162700-1d33-11eb-82c3-2f89e5bffbc5.png)
Now the link in the first column leads to: https://nomad.infra.domain/ui/csi/volumes/test while I think it should lead to https://nomad.infra.domain/ui/csi/volumes/wiki 

The first name (test) is the named as specified in the job file:
```
    volume "test" {
      type      = "csi"
      read_only = false
      source    = "wiki"
    }
```

but the CSI name is the name from the `source` column.

### Reproduction steps

Create a job using a volume that has a different name than it's source

### Job file (if appropriate)

```
job "infra.test" {
  datacenters = ["dc1"]

  group "test" {
    volume "test" {
      type      = "csi"
      read_only = false
      source    = "wiki"
    }


    task "ubuntu" {
      driver = "docker"
      volume_mount {
       volume      = "test"
       destination = "/test"
       read_only   = false
      }

      config {
        image = "ubuntu:20.04"
        command = "sleep"
	args = ["infinity"]
      }
    }

  }
}
```

**Comment by tgross** [comment]
Hi @apollo13! 

I was able to confirm this behavior on the task group page (ex. http://localhost:4646/ui/jobs/example/cache) for both 0.12.5 and the current `master`

I was also able to confirm this behavior on the individual allocation page (ex. http://localhost:4646/ui/allocations/09965ad4-a4b8-261b-6b01-03dcb77b473e ), in which case the screenshot for that section looks like this:

<img width="1235" alt="Screen Shot 2020-11-02 at 2 12 44 PM" src="https://user-images.githubusercontent.com/1409219/97909353-155f8680-1d16-11eb-9f5f-64ea5236fdb9.png">

**Comment by tgross** [comment]
Ok, so I've narrowed down what's happening here. Here's my volume registration:

```hcl
id = "test-volume0"
type = "csi"
external_id = "EXTERNAL_ID"
plugin_id = "hostpath-plugin0"
access_mode = "single-node-writer"
attachment_mode = "file-system"
```

And the relevant part of the jobspec:

```hcl
job "example" {

  group "cache" {
    volume "myvolume" {
      type   = "csi"
      source = "test-volume0"
    }

    task "redis" {
      volume_mount {
        volume      = "myvolume"
        destination = "${NOMAD_ALLOC_DIR}/test"
      }
    }
  }
}
```

But when I hit the allocations API that's populating this data:

```
$ curl -s http://localhost:4646/v1/allocation/:alloc_id | jq '.Job.TaskGroups[0].Volumes'
{
  "myvolume": {
    "MountOptions": null,
    "Name": "myvolume",
    "ReadOnly": false,
    "Source": "test-volume0",
    "Type": "csi"
  }
}
```

So the API is returning `myvolume` as the name of the volume within the task group. But that's not how you query for a volume. These two UI sections should both be populating the URL of the link with the source and not the name of the volume. I think the allocation page has the nicest way of showing that -- it's showing both the bolded name and the link to the real volume, but the link we construct is wrong.

I definitely tested this out a bunch... using a volume name in the task group that was a valid prefix for the real volume name! (ex. `test` is a valid prefix of `test-volume`). 🤦

**Comment by tgross** [comment]
See also: https://github.com/hashicorp/nomad/issues/9297, which suggests the issue is somewhere in the API and not the UI.

I'll self-assign this issue and follow-up if it turns out to coincidentally be a problem in both the CLI and the UI by accident, and I can't manage to do the tiny UI PR myself.

**Comment by tgross** [comment]
I have a patch up for both this and the `nomad alloc status -verbose` fix, which turned out to be coincidentally the same bug in both the CLI and UI and not the API after all.

**Comment by tgross** [comment]
Fixed in #9355 and this will ship in 1.0

**Comment by apollo13** [comment]
Lovely, thanks!
